// @hikionori/admin-sdk/dist/info/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteInfo = exports.updateInfo = exports.createInfo = exports.getAllInfos = exports.getInfoById = exports.InfoBuilder = exports.ContentBuilder = void 0;
const axios_1 = require("axios");
const config_1 = require("../config");
class ContentBuilder {
    setContentType(content_type) {
        this.content_type = content_type;
    }
    setData(data) {
        this.data = data;
    }
    getContentType() {
        return this.content_type;
    }
    getData() {
        return this.data;
    }
}
exports.ContentBuilder = ContentBuilder;
class InfoBuilder {
    setId(id) {
        this.id = id;
    }
    setTitle(title) {
        this.title = title;
    }
    setTheme(theme) {
        this.theme = theme;
    }
    setContentLevels(content_levels) {
        this.content_levels = content_levels;
    }
    getId() {
        return this.id;
    }
    getTitle() {
        return this.title;
    }
    getTheme() {
        return this.theme;
    }
    getContentLevels() {
        return this.content_levels;
    }
}
exports.InfoBuilder = InfoBuilder;
function getInfoById(id) {
    return axios_1.default.get(`${config_1.baseUrl}/admin/get/info?id=${id}`);
}
exports.getInfoById = getInfoById;
function getAllInfos() {
    return axios_1.default.get(`${config_1.baseUrl}/admin/get/info/all`);
}
exports.getAllInfos = getAllInfos;
function createInfo(info) {
    return axios_1.default.post(`${config_1.baseUrl}/admin/create/info`, info);
}
exports.createInfo = createInfo;
function updateInfo(id, info) {
    return axios_1.default.post(`${config_1.baseUrl}/admin/update/info?id=${id}`, info);
}
exports.updateInfo = updateInfo;
function deleteInfo(id) {
    return axios_1.default.post(`${config_1.baseUrl}/admin/del/info?id=${id}`);
}
exports.deleteInfo = deleteInfo;
